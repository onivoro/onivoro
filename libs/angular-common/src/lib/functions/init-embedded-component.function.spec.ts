import { initEmbeddedComponent } from './init-embedded-component.function';

const apiKey = 'apiKey';
const apiId = 'apiId';

const setup = () => {
    const slot = getTestElement();
    const mockCredentialHandler = jest.fn();

    return { slot, mockCredentialHandler };
};

describe('initEmbeddedComponent', () => {
    it('sets the api credentials', () => {
        const { slot, mockCredentialHandler } = setup();
        initEmbeddedComponent(slot, mockCredentialHandler);
        expect(mockCredentialHandler).toHaveBeenCalledWith({ apiId, apiKey });
    });

    it('pulls specified attributes from the element', () => {
        const { slot, mockCredentialHandler } = setup();
        const attributes = initEmbeddedComponent(slot, mockCredentialHandler, { id: '123', accentContrast: 'hotpink', notDefined: 'should not change' });
        expect(attributes).toMatchSnapshot();
    });
});

function getTestElement() {
    const slot = document.createElement('div');
    slot.setAttribute('id', 'ui-answers');
    slot.setAttribute('primary', '#58a4b4');
    slot.setAttribute('accent', '#411910');
    slot.setAttribute('primary-contrast', '#ffffff');
    slot.setAttribute('accent-contrast', '#ffffff');
    slot.setAttribute('api-key', apiKey);
    slot.setAttribute('api-id', apiId);
    return slot;
}