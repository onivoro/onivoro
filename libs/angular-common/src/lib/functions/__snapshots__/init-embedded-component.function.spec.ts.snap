// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initEmbeddedComponent pulls specified attributes from the element 1`] = `
Object {
  "accentContrast": "#ffffff",
  "id": "ui-answers",
  "notDefined": "should not change",
}
`;
